### PR TITLE
Documentation of metadata class

### PR DESCRIPTION
**Pull request by rpitonak**

### CODE REVIEW

File: conu/apidefs/metadata.py [hunk 1]
```diff
@@ -8,6 +8,15 @@ class Metadata(object):
 
     def __init__(self, name=None, identifier=None, labels=None, command=None,
                  creation_timestamp=None, env_variables=None):
+        """
+        :param name: str, name of object
+        :param identifier: int, id of object
```
Comment (TomasTomecek) [inline conu/apidefs/metadata.py]: most of the time it's hex, so please `str`

File: conu/apidefs/metadata.py [hunk 2]
```diff
@@ -8,6 +8,15 @@ class Metadata(object):
...
+        :param labels: dict, {key: value}
+        :param command: str, command to be executed
```
Comment (TomasTomecek) [inline conu/apidefs/metadata.py]: this is worth debating: `str` or a `list` of `str`s?

File: conu/apidefs/metadata.py [hunk 3]
```diff
@@ -8,6 +8,15 @@ class Metadata(object):
...
+        :param creation_timestamp: str, creation time of object instance
```
Comment (TomasTomecek) [inline conu/apidefs/metadata.py]: +1, metadata class would then parse it to datetime object

File: conu/apidefs/metadata.py [hunk 4]
```diff
@@ -23,15 +32,32 @@ class ContainerMetadata(Metadata):
 
     def __init__(self, name=None, identifier=None, ports=None, labels=None, command=None,
                  creation_timestamp=None, env_variables=None,
-                 image=None, hostname=None, ipv4=None, ipv6=None, status=None):
+                 image=None, hostname=None, ipv4_addresses=None, ipv6_addresses=None, status=None):
+        """
+        :param name: str, name of container
+        :param identifier: int, id of container
```
Comment (TomasTomecek) [inline conu/apidefs/metadata.py]: the same

File: conu/apidefs/metadata.py [hunk 5]
```diff
@@ -23,15 +32,32 @@ class ContainerMetadata(Metadata):
...
+        :param ports: list, list of ports
```
Comment (TomasTomecek) [inline conu/apidefs/metadata.py]: is this a list of exposed ports?

File: conu/apidefs/metadata.py [hunk 6]
```diff
@@ -23,15 +32,32 @@ class ContainerMetadata(Metadata):
...
+        :param labels: dict, {key: value}
+        :param command: str, command to be executed
+        :param creation_timestamp: str, creation time of container
+        :param env_variables: dict, {name: value}
+        :param image: str, name of the image
```
Comment (TomasTomecek) [inline conu/apidefs/metadata.py]: would be awesome to link to real image then

File: conu/apidefs/metadata.py [hunk 7]
```diff
@@ -42,6 +68,21 @@ class ImageMetadata(Metadata):
 
     def __init__(self, name=None, identifier=None, ports=None, labels=None, command=None, creation_timestamp=None,
                  env_variables=None, image_names=None):
+        """
+        :param name: str, name of image
+        :param identifier: int, id of image
+        :param ports: list, list of ports
```
Comment (TomasTomecek) [inline conu/apidefs/metadata.py]: what's the semantics here?

File: conu/apidefs/metadata.py (hunk 5 above)
Comment (rpitonak) [reply conu/apidefs/metadata.py]: Yes, the idea is to have list of exposed ports and then also dictionary which looks like {"ip_address": "port"}, Does it make sense to you? Or is enough to have IP addresses in list?

File: conu/apidefs/metadata.py (hunk 2 above)
Comment (rpitonak) [reply conu/apidefs/metadata.py]: +1 for `list` of `str`s : 
- `["python3", "-m", "http.server", "--bind", "0.0.0.0", "8080"]`

Comment (TomasTomecek) [reply conu/apidefs/metadata.py]: yup, we can even easily transform to plain str if some backend uses such syntax; going from str to list of strs is much harder

File: conu/apidefs/metadata.py (hunk 5 above)
Comment (TomasTomecek) [reply conu/apidefs/metadata.py]: +1 on having those as 2 separate fields

Comment (TomasTomecek) [reply conu/apidefs/metadata.py]: e.g. `exposed_ports` and `port_mappings`

**Pull request merged**